src/viewer.ts

```diff
@@ -827,33 +812,36 @@ class Viewer {
         // reset animation state
         this.animTracks = [];
         this.animationMap = { };
-        this.observer.set('animation.list', '[]');
-        this.observer.set('scene.variants.list', '[]');
-
-        this.observer.set('morphs', null);
-
-        this.updateSceneInfo();
-
-        this.dirtyWireframe = this.dirtyBounds = this.dirtySkeleton = this.dirtyGrid = this.dirtyNormals = true;
-        this.renderNextFrame();
     }
 
     updateSceneInfo() {
         let meshCount = 0;
+        let meshVRAM = 0;
         let vertexCount = 0;
         let primitiveCount = 0;
+        let materialCount = 0;
+        let textureCount = 0;
+        let textureVRAM = 0;
         let variants: string[] = [];
 
         // update mesh stats
         this.assets.forEach((asset) => {
             variants = variants.concat(asset.resource.getMaterialVariants());
             asset.resource.renders.forEach((renderAsset: Asset) => {
+                meshCount += renderAsset.resource.meshes.length;
                 renderAsset.resource.meshes.forEach((mesh: Mesh) => {
-                    meshCount++;
                     vertexCount += mesh.vertexBuffer.getNumVertices();
                     primitiveCount += mesh.primitive[0].count;
+                    meshVRAM += mesh.vertexBuffer.numBytes + (mesh.indexBuffer?.numBytes ?? 0);
                 });
             });
+
+            materialCount += asset.resource.materials.length;
+
+            textureCount += asset.resource.textures.length;
+            asset.resource.textures.forEach((texture: Asset) => {
+                textureVRAM += texture.resource.gpuSize;
```

Comment:
Could textures be shared between assets here?

Reply:
Yes but can they reference the same external images? Or would we load them multiple times in that case as well.

Reply:
I guess you show stats per glb anyways, so that would make a difference.